src/Artifacts/FileSystem.cs

```diff
@@ -59,5 +66,27 @@ public bool FileExists(FileInfo file)
         {
             return file.Exists;
         }
+
+        public bool TryCloneFile(FileInfo sourceFile, FileInfo destinationFile)
+        {
+#if NETSTANDARD2_0_OR_GREATER
+            string sourcePath = sourceFile.FullName;
+            string destPath = destinationFile.FullName;
+            if (CoW.CopyOnWriteLinkSupportedBetweenPaths(sourcePath, destPath))
```

Comment:
Most users this will be false 100% of the time. Is there a perf hit? I implemented a different approach to avoid checking each time. If this is truly no perf hit we should change the other SDK to match this approach (which is better). But given the perf requirements, I would guess this would be a perf hit on large repos.

Reply:
The CoW package has a highly tuned, cached lookup table to minimize time to make this call (as I expected it to be used on most calls as we're doing here). It's more complete (handles more cases) than the drive letter cache in the CoW SDK in this repo. I can migrate that logic to use this.